pywren_ibm_cloud/libs/ibm/cloudfunctions_client.py

```diff
@@ -192,56 +190,65 @@ def create_package(self, package):
         else:
             logger.debug("OK --> Created package {}".format(package))
 
-    def invoke(self, action_name, payload):
+    def invoke(self, package, action_name, payload={}, self_invoked=False):
         """
         Invoke an IBM Cloud Function by using new request.
         """
         exec_id = payload['executor_id']
         call_id = payload['call_id']
-        url = '/'.join([self.endpoint, 'api', 'v1', 'namespaces', self.effective_namespace, 'actions', self.package, action_name])
-        url = urlparse(url)
+        url = '/'.join([self.endpoint, 'api', 'v1', 'namespaces', self.effective_namespace, 'actions', package, action_name])
+        parsed_url = urlparse(url)
         start = time.time()
         try:
-            ctx = ssl._create_unverified_context()
-            conn = http.client.HTTPSConnection(url.netloc, context=ctx)
-            conn.request("POST", url.geturl(),
-                         body=json.dumps(payload),
-                         headers=self.headers)
-            resp = conn.getresponse()
-            data = resp.read()
+            if self.is_cf_cluster:
+                resp = self.session.post(url, json=payload)
+                resp_status = resp.status_code
+                data = resp.json()
+            else:
+                ctx = ssl._create_unverified_context()
+                conn = http.client.HTTPSConnection(parsed_url.netloc, context=ctx)
+                conn.request("POST", parsed_url.geturl(),
+                             body=json.dumps(payload),
+                             headers=self.headers)
+                resp = conn.getresponse()
+                resp_status = resp.status
+                data = json.loads(resp.read().decode("utf-8"))
+                conn.close()
         except Exception as e:
-            conn.close()
-            logger.debug(str(e))
-            return self.request_invoke(action_name, payload)
+            if not self.is_cf_cluster:
+                conn.close()
+            log_msg = ('ExecutorID {} - Function {} invocation failed: {}'.format(exec_id, call_id, str(e)))
+            logger.debug(log_msg)
+            if self_invoked:
+                return None
+            return self.invoke(action_name, payload, self_invoked=True)
```

Comment:
Missing `package` first parameter on this line

Reply:
@JosepSampe ? ;)

Reply:
Fixed